File: leetcode/src/java/PascalsTriangle.java
import java.util.ArrayList;
import java.util.List;

public class PascalsTriangle {
    public static void main(String[] args) {
        PascalsTriangle instance = new PascalsTriangle();
        List<List<Integer>> result = instance.generate(5);
        for (List<Integer> item : result) {
            System.out.println(item);
        }
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> list = new ArrayList<Integer>();
            if (i == 0) {
                list.add(1);
            } else {
                List<Integer> lastList = result.get(result.size() - 1);
                list.add(lastList.get(0));
                for (int j = 0; j < lastList.size() - 1; j++) {
                    list.add(lastList.get(j) + lastList.get(j + 1));
                }
                list.add(lastList.get(lastList.size() - 1));
            }
            result.add(list);
        }
        return result;
    }
}

File: leetcode/src/python/pascals_triangle.py
from typing import List

class PascalsTriangle:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        if numRows >= 1:
            result.append([1])
        if numRows >= 2:
            result.append([1, 1])
        if numRows >= 3:
            last_list = result[len(result) - 1]
            for i in range(numRows - 2):
                current_list = [1]
                for j in range(len(last_list) - 1):
                    current_list.append(last_list[j] + last_list[j + 1])
                current_list.append(1)
                result.append(current_list)
                last_list = current_list
        return result

if __name__ == '__main__':
    instance = PascalsTriangle()
    numRows = 5
    print(instance.generate(numRows))
